ksqldb-rest-app/src/test/java/io/confluent/ksql/api/integration/ApiIntegrationTest.java

```diff
@@ -659,6 +660,136 @@ private void shouldFailToExecuteQuery(final String sql, final String message) {
+
+    // Create a write stream to capture the incomplete response
+    ReceiveStream writeStream = new ReceiveStream(vertx);
+
+    // Make the request to stream a print
+    JsonObject printProperties = new JsonObject().put("auto.offset.reset", "earliest");
+    JsonObject printRequestBody = new JsonObject().put("sql", sql)
+        .put("properties", printProperties);
+    VertxCompletableFuture<HttpResponse<Void>> responseFuture = new VertxCompletableFuture<>();
+
+    // When:
+    client.post("/query-stream").as(BodyCodec.pipe(writeStream))
+        .sendJsonObject(printRequestBody, responseFuture);
+
+    // Then:
+    AtomicReference<PrintResponse> atomicReference = new AtomicReference<>();
+    assertThatEventually(() -> {
+      try {
+        Buffer buff = writeStream.getBody();
+        PrintResponse printResponse = new PrintResponse(buff.toString());
+        atomicReference.set(printResponse);
+        return printResponse.getRows().size();
+      } catch (Throwable t) {
+        return -1;
+      }
+    }, is(1));
+
+    PrintResponse printResponse = atomicReference.get();
+    assertThat(printResponse.getRows().get(0), containsString(
+        "key: {\"F1\":[\"a\"]}, value: {\"STR\":\"FOO\",\"LONG\":1,\"DEC\":1.11,\"BYTES_\":\"AQ==\",\"ARRAY\":[\"a\"],\"MAP\":{\"k1\":\"v1\"},\"STRUCT\":{\"F1\":2},\"COMPLEX\":{\"DECIMAL\":0.0,\"STRUCT\":{\"F1\":\"v0\",\"F2\":0},\"ARRAY_ARRAY\":[[\"foo\"]],\"ARRAY_STRUCT\":[{\"F1\":\"v0\"}],\"ARRAY_MAP\":[{\"k1\":0}],\"MAP_ARRAY\":{\"k\":[\"v0\"]},\"MAP_MAP\":{\"k\":{\"k\":0}},\"MAP_STRUCT\":{\"k\":{\"F1\":\"v0\"}}},\"TIMESTAMP\":1,\"DATE\":1,\"TIME\":0}, partition: 0"));
+  }
+
+  @Test
+  public void shouldExecutePrintQueryNoLimit() {
+    // Given:
+    String sql = "PRINT " + TEST_TOPIC + ";";
+
+    // Create a write stream to capture the incomplete response
+    ReceiveStream writeStream = new ReceiveStream(vertx);
+
+    // Make the request to stream a print
+    JsonObject printProperties = new JsonObject().put("auto.offset.reset", "earliest");
+    JsonObject printRequestBody = new JsonObject().put("sql", sql)
+        .put("properties", printProperties);
+    VertxCompletableFuture<HttpResponse<Void>> responseFuture = new VertxCompletableFuture<>();
+
+    // When:
+    client.post("/query-stream").timeout(Long.MAX_VALUE).as(BodyCodec.pipe(writeStream))
+        .sendJsonObject(printRequestBody, responseFuture);
+
+    // Then:
+    AtomicReference<PrintResponse> atomicReference = new AtomicReference<>();
+    assertThatEventually(() -> {
+      try {
+        Buffer buff = writeStream.getBody();
+        PrintResponse printResponse = new PrintResponse(buff.toString());
+        atomicReference.set(printResponse);
+        return printResponse.getRows().size();
+      } catch (Throwable t) {
+        return -1;
+      }
+    }, greaterThanOrEqualTo(6));
+
+    PrintResponse printResponse = atomicReference.get();
+    assertThat(writeStream.isEnded(), is(false));
```

Comment:
This is a good check to have. Can we add the equivalent for the situations in which we expect the response stream to be ended? Namely, when the PRINT TOPIC has a LIMIT clause, and also when there is no LIMIT clause but the client closes the connection.

Reply:
I added this check to the existing shouldExecutePrint test. 

About the test closing the connection, I tried to find a way to simulate this with Vertx's WebClient, but couldn't really find anything. Do you have a suggestion on how to do this?

Reply:
> About the test closing the connection, I tried to find a way to simulate this with Vertx's WebClient, but couldn't really find anything. Do you have a suggestion on how to do this?

I think closing the client should work? If it's too annoying to implement this, we might be OK without it.

Reply:
Closing only the client didn't also work, but I found a workaround to test this and pushed the test "shouldTryToPrintAndSendDoneResponseIfConnectionClosed".